### ISSUE TITLE
Warnings using django 1.8.1

### ISSUE THREAD

**Issue by paridin** [vijaykatam/django-cache-manager#15]
i got this warnings using django 1.8.1

**mixins.py**
```
..../python3.4/site-packages/django_cache_manager/model_cache_sharing/backends/shared_memory.py:28: RemovedInDjango19Warning: 'get_cache' is deprecated in favor of 'caches'.
  self._cache_backend = get_cache(_cache_name)
```
**model_cache_sharing/backends/shared_memory.py**
```
.../python3.4/site-packages/django_cache_manager/mixins.py:91: RemovedInDjango19Warning: 'get_cache' is deprecated in favor of 'caches'.
  self._cache_backend = get_cache(_cache_name)
```
You can be solved by doing this, but it affect the core?

**mixins.py**
```
line 6:
- from django.core.cache import get_cache
+ from django.core.cache import caches

line 91:
- self._cache_backend = get_cache(_cache_name)
+ self._cache_backend = caches[_cache_name]
```
**model_cache_sharing/backends/shared_memory.py**
```
line 4:
- from django.core.cache import get_cache
+ from django.core.cache import caches

line 28:
- self._cache_backend = get_cache(_cache_name)
+ self._cache_backend = caches[_cache_name]
```

I will appreciate your comments.

**Comment by vijaykatam** [comment]
Prior to 1.8 Django's API(get_cache) for getting a handle to the cache backend is not thread safe so had to be called anytime something needed it instead of initializing it once per module. The newer API caches is thread safe and can be used at the module level. 

Although I don't see any issues with using the older cache API, this component is not officially ready for use in django 1.8, if you notice the travis builds are against django 1.5, 1.6 and 1.7, I have a few changes in mind that should remove dependencies on django model private API's and also improve efficiency of cache eviction while also helping transition to support django 1.8 easier. Until then I wouldn't recommend using this for django 1.8 in production environments.

**Comment by ajenhl** [comment]
Is there any help I can give you to get the code usable for 1.8?

**Comment by vijaykatam** [comment]
I enabled [travis build](https://travis-ci.org/vijaykatam/django-cache-manager) for django 1.8 and there are couple of failing tests. Feel free to look- https://github.com/vijaykatam/django-cache-manager/tree/django_1.8

**Comment by ajenhl** [comment]
It looks as though following https://docs.djangoproject.com/en/1.8/ref/models/meta/#migrating-old-meta-api should help here, changing models.py so that invalidate_model_cache specifies:

```python
related_tables = set([f.related_model._meta.db_table for f in sender._meta.get_fields()
                      if (f.one_to_many or f.one_to_one or f.many_to_one)
                      or (f.many_to_many and not f.auto_created)])
```

rather than

```python
related_tables = set([rel.model._meta.db_table for rel in sender._meta.get_all_related_objects()])
# temporary fix for m2m relations with an intermediate model, goes away after better join caching
related_tables |= set([field.rel.to._meta.db_table for field in sender._meta.fields if issubclass(type(field), RelatedField)]) 
```
All of the tests pass, though I'm a little curious about the test comments saying "Because of m2m fix the no. of queries will be 2 instead of 1" - *should* the number of queries be 2, or would it ideally be 1?

I haven't created a pull request because I haven't added any logic to check whether the Django version to use the new API.

**Comment by ajenhl** [comment]
Would creating a pull request with the following code be helpful in getting this fixed?

```python
if django.get_version() >= '1.8':
    related_tables = set([f.related_model._meta.db_table for f in sender._meta.get_fields()
                          if (f.one_to_many or f.one_to_one or f.many_to_one)
                          or (f.many_to_many and not f.auto_created)])
else:
    related_tables = set([rel.model._meta.db_table for rel in sender._meta.get_all_related_objects()])
    # temporary fix for m2m relations with an intermediate model, goes away after better join caching
    related_tables |= set([field.rel.to._meta.db_table for field in sender._meta.fields if issubclass(type(field), RelatedField)]) 
```

**Comment by vijaykatam** [comment]
The replacement for get_all_related_objects() in 1.8 according to [doc](https://docs.djangoproject.com/en/1.8/ref/models/meta/#migrating-from-the-old-api) is 
```python
[
    f for f in MyModel._meta.get_fields()
    if (f.one_to_many or f.one_to_one) and f.auto_created
]
```

**Comment by ajenhl** [comment]
Indeed, but using that results in six test failures. That expression does not return many to many fields. But you're right that the code should probably be:

```python
related_tables = set(
    [f.related_model._meta.db_table for f in sender._meta.get_fields()
     if ((f.one_to_many or f.one_to_one) and f.auto_created)
     or f.many_to_one or (f.many_to_many and not f.auto_created)])
```

**Comment by vijaykatam** [comment]
go ahead and submit a pull with django version check

**Comment by vijaykatam** [comment]
Released [v0.3](https://github.com/vijaykatam/django-cache-manager/releases/tag/v0.3) with help from @ajenhl
